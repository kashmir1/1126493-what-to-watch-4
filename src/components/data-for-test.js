export const moviePoster = {
  id: 1111,
  bg: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  previewImg: `img/the-grand-budapest-hotel-poster.jpg`,
  src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  title: `The Grand Budapest Hotel`,
  genre: `Drame`,
  year: 2014,
  time: 99,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  rating: 4.9,
  votes: 468,
  director: `Wes Anderson`,
  starring: [`Ralph Fiennes`, `F. Murray Abraham`, `Mathieu Amalric`],
};

export const films = [{
  id: 2298,
  bg: `https://loremflickr.com/1300/550`,
  poster: `img/snatch.jpg`,
  previewImg: `img/snatch.jpg`,
  src: `https://upload.wikimedia.org/wikipedia/commons/7/7d/Kettlebell_Open_Palm_Snatch.webm`,
  previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  title: `Snatch`,
  genre: `Crime`,
  year: 2000,
  time: 104,
  description: `Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers and supposedly Jewish jewelers fight to track down a priceless stolen diamond.`,
  rating: 9.8,
  votes: 842,
  director: `Guy Ritchie`,
  starring: [`Jason Statham`, `Brad Pitt`, `Benicio Del Toro`],
},
{
  id: 2245,
  bg: `https://loremflickr.com/1300/550`,
  poster: `img/moonrise-kingdom.jpg`,
  previewImg: `img/moonrise-kingdom.jpg`,
  previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  src: `https://upload.wikimedia.org/wikipedia/commons/1/14/Aurora_Borealis_in_the_UK.webm`,
  title: `Moonrise Kingdom`,
  genre: `Romance`,
  year: 2012,
  time: 94,
  description: `A pair of young lovers flee their New England town, which causes a local search party to fan out to find them.`,
  rating: 7.9,
  votes: 529,
  director: `Wes Anderson`,
  starring: [`Jared Gilman`, `Kara Hayward`, `Bruce Willis`],
},
{
  id: 2249,
  bg: `https://loremflickr.com/1300/550`,
  poster: `img/seven-years-in-tibet.jpg`,
  previewImg: `img/seven-years-in-tibet.jpg`,
  previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  src: `https://upload.wikimedia.org/wikipedia/commons/9/93/Tsering_Woeser.webm`,
  title: `Seven Years in Tibet`,
  genre: `Drame`,
  year: 1997,
  time: 136,
  description: `True story of Heinrich Harrer, an Austrian mountain climber who became friends with the Dalai Lama at the time of China's takeover of Tibet.`,
  rating: 7.1,
  votes: 661,
  director: `Jean-Jacques Annaud`,
  starring: [`Brad Pitt`, `David Thewlis`, `BD Wong`],
},
{
  id: 9921,
  bg: `https://loremflickr.com/1300/550`,
  poster: `img/midnight-special.jpg`,
  previewImg: `img/midnight-special.jpg`,
  previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  src: `https://upload.wikimedia.org/wikipedia/commons/0/00/Solstice_Timelapse_-_Day_and_Night_in_the_Midnight_Sun.webm`,
  title: `Midnight Special`,
  genre: `Sci-Fi`,
  year: 2016,
  time: 112,
  description: `A father and son go on the run, pursued by the government and a cult drawn to the child's special powers.`,
  rating: 4.7,
  votes: 717,
  director: `Jeff Nichols`,
  starring: [`Michael Shannon`, `Joel Edgerton`, `Kirsten Dunst`],
},
{
  id: 9963,
  bg: `https://loremflickr.com/1300/550`,
  poster: `img/war-of-the-worlds.jpg`,
  previewImg: `img/war-of-the-worlds.jpg`,
  src: `https://upload.wikimedia.org/wikipedia/commons/e/e6/Real_World_Chicago_Protest_Footage.webm`,
  previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  title: `War of the Worlds`,
  genre: `Sci-Fi`,
  year: 2005,
  time: 116,
  description: `As Earth is invaded by alien tripod fighting machines, one family fights for survival in this sci-fi action film.`,
  rating: 7.4,
  votes: 622,
  director: `Steven Spielberg`,
  starring: [`Tom Cruise`, `Dakota Fanning`, `Tim Robbins`],
},
{
  id: 7732,
  bg: `https://loremflickr.com/1300/550`,
  poster: `img/dardjeeling-limited.jpg`,
  previewImg: `img/dardjeeling-limited.jpg`,
  src: `https://upload.wikimedia.org/wikipedia/commons/8/84/JRW_287_Panda_Kuroshio_at_Tennoji_Station_2017-08-04_360p.webm`,
  previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  title: `Dardjeeling Limited`,
  genre: `Comedie`,
  year: 2007,
  time: 91,
  description: `A year after their father's funeral, three brothers travel across India by train in an attempt to bond with each other.`,
  rating: 4.9,
  votes: 468,
  director: `Wes Anderson`,
  starring: [`Owen Wilson`, `Adrien Brody`, `Jason Schwartzman`],
},
{
  id: 7794,
  bg: `https://loremflickr.com/1300/550`,
  poster: `img/orlando.jpg`,
  previewImg: `img/orlando.jpg`,
  src: `https://upload.wikimedia.org/wikipedia/commons/5/5a/V%C3%ADdeo_Ag%C3%AAncia_Nacional_44.webm`,
  previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  title: `Orlando`,
  genre: `Drame`,
  year: 1992,
  time: 94,
  description: `After Queen Elizabeth I commands him not to grow old, a young nobleman struggles with love and his place in the world.`,
  rating: 8.2,
  votes: 941,
  director: `Sally Potter`,
  starring: [`Tilda Swinton`, `Billy Zane`, `Quentin Crisp`],
},
{
  id: 7714,
  bg: `https://loremflickr.com/1300/550`,
  poster: `img/mindhunter.jpg`,
  previewImg: `img/mindhunter.jpg`,
  src: `https://upload.wikimedia.org/wikipedia/commons/a/a8/Hunter_Hayes_Performing_at_the_White_House_%281999%29.webm`,
  previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  title: `Mindhunter`,
  genre: `Thriller`,
  year: 2017,
  time: 60,
  description: `Set in the late 1970s, two FBI agents are tasked with interviewing serial killers to solve open cases.`,
  rating: 5.5,
  votes: 811,
  director: `Joe Penhall`,
  starring: [`Jonathan Groff`, `Holt McCallany`, `Anna Torv`],
}];

export const comments = [{
  id: 1,
  user: {
    id: 4,
    name: `Kate Muir`,
  },
  rating: 8.9,
  comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  date: `2019-05-08T14:13:56.569Z`,
}, {
  id: 1,
  user: {
    id: 4,
    name: `Kate Muir`,
  },
  rating: 8.9,
  comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  date: `2019-05-08T14:13:56.569Z`,
}];
