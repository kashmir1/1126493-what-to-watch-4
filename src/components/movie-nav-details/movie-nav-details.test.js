import React from 'react';
import renderer from 'react-test-renderer';
import MovieNavDetails from './movie-nav-details.jsx';
import {films} from '../data-for-test.js';

const film = films[0];

describe(`MovieNavDetails`, () => {
  it(`Render MovieNavDetails time more hour`, () => {
    const tree = renderer.create(
        <MovieNavDetails
          director={film.director}
          genre={film.genre}
          starring={film.starring}
          time={80}
          year={film.year}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render MovieNavDetails time less hour`, () => {
    const tree = renderer.create(
        <MovieNavDetails
          director={film.director}
          genre={film.genre}
          starring={film.starring}
          time={30}
          year={film.year}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render MovieNavDetails time less minute`, () => {
    const tree = renderer.create(
        <MovieNavDetails
          director={film.director}
          genre={film.genre}
          starring={film.starring}
          time={0}
          year={film.year}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
