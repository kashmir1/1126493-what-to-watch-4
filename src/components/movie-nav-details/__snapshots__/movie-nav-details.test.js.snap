// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieNavDetails Render MovieNavDetails 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        Guy Ritchie
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <span
        className="movie-card__details-value"
      >
        Jason Statham, 
Brad Pitt, 
Benicio Del Toro
      </span>
    </p>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
      >
        1h 44m
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        Crime
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      >
        2000
      </span>
    </p>
  </div>
</div>
`;
